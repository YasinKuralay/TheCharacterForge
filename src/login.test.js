import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Login from "./login.js";

it("renders the Login component without crashing", () => {
    const div = document.createElement("div");
    // The BrowserRouter is for eliminating the unnecessary error "Don't use Link outside Router" from Jest, which is caused by using an old version of the Router, and will be fixed
    ReactDOM.render(
        <BrowserRouter>
            <Login />
        </BrowserRouter>,
        div
    );
    ReactDOM.unmountComponentAtNode(div);
});

it("matches the previous snapshot of the Login Component", () => {
    const LoginComponentSnapshot = renderer
        .create(
            <BrowserRouter>
                <Login />
            </BrowserRouter>
        )
        .toJSON();
    expect(LoginComponentSnapshot).toMatchSnapshot();
});
