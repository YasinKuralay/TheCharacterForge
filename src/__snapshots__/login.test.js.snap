// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the previous snapshot of the Login Component 1`] = `
<div
  id="register"
>
  <h2>
    Login If You Have an Account
  </h2>
  <input
    autoComplete="off"
    name="email"
    onChange={[Function]}
    placeholder="email"
    type="email"
  />
  <input
    autoComplete="off"
    name="password"
    onChange={[Function]}
    placeholder="password"
    type="password"
  />
  <button
    onClick={[Function]}
  >
    Login
  </button>
  <p
    id="forgotPassword"
  >
    Forgot your
     
    <a
      href="/ResetPassword"
      id="textDecoration"
      onClick={[Function]}
    >
      password?
    </a>
  </p>
  <p
    id="ifYou"
  >
    If you don't have an account, please
     
    <a
      href="/"
      id="textDecoration"
      onClick={[Function]}
    >
      register
    </a>
     
    first.
  </p>
</div>
`;
